src/phoenix/server/api/routers/v1/spans.py

```diff
@@ -1119,3 +1121,129 @@ def convert_api_span_for_insertion(api_span: Span) -> SpanForInsertion:
+            "it cannot contain slash (/), question mark (?), or pound sign (#) characters."
+        )
+    ),
+    span_identifier: str = Path(
+        description="The span identifier: either a relay GlobalID or OpenTelemetry span_id"
+    ),
+) -> None:
+    """
+    Delete a span by span_identifier within a project, leaving all descendants alone.
+
+    1. Find the project by identifier
+    2. Find the target span to delete
+    3. Delete only the target span (leave descendants alone)
+    4. Check if trace is empty - if so, delete the trace record
+    5. If span had a parent, propagate negative cumulative values up ancestor chain
+    6. Return 204 No Content on success
+    """
+    async with request.app.state.db() as session:
+        project = await _get_project_by_identifier(session, project_identifier)
+        project_id: int = project.id
+
+        # Try to parse as GlobalID first, then fall back to span_id
+        try:
+            span_rowid = from_global_id_with_expected_type(
+                GlobalID.from_id(span_identifier),
+                "Span",
+            )
+            # Find by database rowid and verify it belongs to the project
+            target_span = await session.scalar(
+                select(models.Span)
+                .join(models.Trace, models.Trace.id == models.Span.trace_rowid)
+                .where(models.Trace.project_rowid == project_id)
+                .where(models.Span.id == span_rowid)
+            )
+            error_detail = (
+                f"Span with relay ID '{span_identifier}' not found in "
+                f"project '{project_identifier}'"
+            )
+        except Exception:
+            # Find by OpenTelemetry span_id
+            target_span = await session.scalar(
+                select(models.Span)
+                .join(models.Trace, models.Trace.id == models.Span.trace_rowid)
+                .where(models.Trace.project_rowid == project_id)
+                .where(models.Span.span_id == span_identifier)
+            )
+            error_detail = (
+                f"Span with span_id '{span_identifier}' not found in project '{project_identifier}'"
+            )
+        if target_span is None:
+            raise HTTPException(
+                status_code=HTTP_404_NOT_FOUND,
+                detail=error_detail,
+            )
+
+        # Store values needed for later operations
+        trace_id = target_span.trace_rowid
+        parent_id = target_span.parent_id
+        cumulative_error_count = target_span.cumulative_error_count
+        cumulative_llm_token_count_prompt = target_span.cumulative_llm_token_count_prompt
+        cumulative_llm_token_count_completion = target_span.cumulative_llm_token_count_completion
+
+        # Delete only the target span (leave descendants alone)
+        await session.execute(sa_delete(models.Span).where(models.Span.id == target_span.id))
```

Comment:
I think you can do it this way instead of querying twice. The `predicate` here would then be defined above to be either `models.Span.id == ...` or `models.Span.span_id == ...`
```suggestion
        span = await session.scalar(sa.delete(models.Span).where(predicate).returning(models.Span))
```

Reply:
agree, was just trying to avoid if relayid conflict with otelId but I don't think that would be true